src/lib/parser.c

```diff
@@ -744,6 +750,40 @@ static inline RdbStatus unpackList(RdbParser *p, unsigned char *lp) {
     return RDB_STATUS_OK;
 }
 
+/* callback for ziplistValidateIntegrity.
+ * The ziplist element pointed by 'p' will be converted and stored into listpack. */
+static int ziplistEntryConvertAndValidate(unsigned char *p, unsigned int head_count, void *userdata) {
```

Comment:
This code is aligned with redis integrated parser (more below). IMHO as long as we have in mind that this parser might replace integrated parser in the future, we better keep it aligned. Otherwise we should have more adequate names.

Reply:
ohh, i think i missed the "convert" part of the function name.
however:
1. i don't think the parser needs to stay in par with redis internals (in the way it operates or the code stricture), if / when we someday replace the parser in redis with the new one, we'll just drop the old.
2. i don't think it's the parser's responsibility to do any conversion. as a parser it should just "parse". if we also want to give it "validation" capabilities, that's (arguably) ok, but it should certainly not be the one that converts ziplist to listpack. instead the app can register to get either a raw ziplist or get the unzipped items (and it's the app's responsibility to create a listpack from it)